src/render_engine/vulkan/vulkan_utils.hpp

```diff
@@ -9,52 +9,28 @@
-#define ERROR_CASE(r)                                                                                                                      \
-    case VK_##r:                                                                                                                           \
-        return #r
-            switch(result) {
-                ERROR_CASE(SUCCESS);
-                ERROR_CASE(NOT_READY);
-                ERROR_CASE(TIMEOUT);
-                ERROR_CASE(EVENT_SET);
-                ERROR_CASE(INCOMPLETE);
-                ERROR_CASE(ERROR_OUT_OF_HOST_MEMORY);
-                ERROR_CASE(ERROR_OUT_OF_DEVICE_MEMORY);
-                ERROR_CASE(ERROR_INITIALIZATION_FAILED);
-                ERROR_CASE(ERROR_DEVICE_LOST);
-
-                default:
-                    return std::string("UNKNOWN_ERROR");
-            }
-#undef ERROR_CASE
-        }
-    };
+namespace nova::renderer {
+    std::string vk_result_to_string(VkResult result);
 
     std::string to_string(VkObjectType obj_type);
 
     std::vector<VkVertexInputBindingDescription>& get_vertex_input_binding_descriptions();
 
     std::vector<VkVertexInputAttributeDescription>& get_vertex_input_attribute_descriptions();
-} // namespace nova::renderer::vulkan
+} // namespace nova::renderer
 
 // Only validate errors in debug mode
 // Release mode needs to be fast A F
 #ifndef NDEBUG
 // NOLINTNEXTLINE(cppcoreguidelines-macro-usage)
-#define NOVA_THROW_IF_VK_ERROR(expr, exception)                                                                                            \
-    do {                                                                                                                                   \
+#define NOVA_CHECK_RESULT(expr)                                                                                                            \
+    {                                                                                                                                      \
         const VkResult result = (expr);                                                                                                    \
         if(result != VK_SUCCESS) {                                                                                                         \
-            throw exception(std::string(__FILE__) + ":" + std::to_string(__LINE__) + "=> " +                                               \
-                            ::nova::renderer::vulkan::vulkan_utils::vk_result_to_string(result));                                          \
+            NOVA_LOG(ERROR) << __FILE__ << ":" << __LINE__ << "=> " << #expr << "=" << vk_result_to_string(result);                        \
         }                                                                                                                                  \
-    } while(false);
+    }
 #else
-#define NOVA_THROW_IF_VK_ERROR(expr, exception) expr
+#define NOVA_CHECK_ERROR(expr) expr
```

Comment:
NOVA_CHECK_ERROR is unused and compilation fail when CMAKE_BUILD_TYPE=Release.
Isn't it supposed to have same name when debug is disabled?

Reply:
Yes - good catch!